invenio_rdm_records/records/jsonschemas/records/record-v1.0.0.json

```diff
@@ -1,402 +1,714 @@
-                "type": "array",
-                "uniqueItems": true,
-                "items": {
-                    "type": "object",
-                    "properties": {
-                      "name": {"type": "string"},
-                      "identifiers": {
-                        "type": "object",
-                        "additionalProperties": {
-                          "type": "string"
-                        }
-                      }
-                    },
-                    "required": ["name", "identifiers"]
-                }
-              }
+            "name": {"type": "string"},
+            "identifiers": {"$ref": "#/definitions/identifiers"}
           },
-          "required": ["name"]
+          "required": ["name", "identifiers"]
+      }
+    },
+
+    "file": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "A file object.",
+      "properties": {
+        "checksum": {
+          "description": "Checksum of the file.",
+          "type": "string"
+        },
+        "size": {
+          "description": "Size of the file in bytes.",
+          "type": "number"
+        },
+        "key": {
+          "description": "Key (filename) of the file.",
+          "type": "string"
+        },
+        "ext": {
+          "description": "File extension.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description for file.",
+          "type": "string"
+        },
+        "order": {
+          "description": "Value to sort according to.",
+          "type": "string"
+        },
+        "default_preview": {
+          "description": "Set to true to use as the default previewed file.",
+          "type": "boolean"
+        },
+        "bucket": {
+          "description": "TODO - Bucket id - what about third party storage systems?",
+          "type": "string"
+        },
+        "file_id": {
+          "description": "TODO - Identifier of the file. - same as for bucket",
+          "type": "string"
         }
-      },
-      "dates": {
-        "description": "Date or date interval.",
-        "minItems": 0,
-        "type": "array",
-        "items": {
-          "type": "object",
-          "additionalProperties": false,
-          "properties": {
-            "description": {
-              "description": "Description of the date or date interval e.g. 'Accepted' or 'Available' (CV).",
-              "type": "string"
-            },
-            "end": {
-              "description": "End date.",
-              "type": "string",
-              "format": "date-time"
-            },
-            "start": {
-              "description": "Start date.",
-              "type": "string",
-              "format": "date-time"
-            },
-            "type": {
-              "description": "Type of the date interval."
-            }
-          },
-          "required": ["type"]
+      }
+    },
+
+    "agent-user": {
```

Comment:
question: agent-user? what is the purpose? It is a bit confusing with user-agent.